services/autoscaling/src/simcore_service_autoscaling/services.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import math
+from datetime import datetime, timedelta
+
+from .core.settings import AwsSettings
+from .utils_aws import AWS_EC2, compose_user_data, start_instance_aws
+from .utils_docker import check_tasks_resources, eval_cluster_resources, need_resources
+
+logger = logging.getLogger(__name__)
+
+
+async def check_dynamic(settings: AwsSettings):
+    """
+    Check if the swarm need to scale up
+
+    """
+    user_data = compose_user_data(settings.AUTOSCALING_AWS)
+
+    # We need the data of each task and the data of each node to know if we need to scale up or not
+    # Test if some tasks are in a pending mode because of a lack of resources
+    resources_are_scarce = await need_resources()
+
+    # We compile RAM and CPU capabilities of each node who have the label sidecar
+    # Total resources of the cluster
+    if resources_are_scarce:
+        total_nodes = await eval_cluster_resources()
+        total_tasks = await check_tasks_resources(total_nodes.nodes_ids)
+        available_cpus = total_nodes.total_cpus - total_tasks.total_cpus_running_tasks
+        available_ram = total_nodes.total_ram - total_tasks.total_ram_running_tasks
+
```

Comment:
In the midterm, we'd also need to check for VRAM if I remember the last talk with MAG correctly